dev/src/lobster/parser.h

```diff
@@ -612,8 +612,13 @@ struct Parser {
                     dest = st.NewSpecUDT(dest->spec_udt->udt);
                     if (dest->spec_udt->udt->is_generic) dest->spec_udt->is_generic = true;
                     for (;;) {
-                        auto s = ParseType(false);
-                        if (st.IsGeneric(s)) dest->spec_udt->is_generic = true;
+                        UnresolvedTypeRef s;
+                        if (allow_wildcard_generic && IsNext(T_QUESTIONMARK)) {
```

Comment:
maybe lift this check out of the `for`, such that only `T<?>` is legal, but not `T<?,U>`, `T<U,?>` and other cases we can't deal with in the typechecker are illegal. That also disallows `T<?,?>` which is probably fine, the `?` can indicate multiple elements.